### ISSUE TITLE
Flaky test suite

### ISSUE THREAD

**Issue by neetopia** [android/kotlin#40]
testJavaModifiers is failing when running the whole test suite, but passing for running single test case. Possibly due to compiled Java classes from other tests not cleared after test case.

**Comment by ting-yuan** [comment]
Besides `testJavaModifiers`, `testAllFunctions` is also failing when running the whole test suite.

**Comment by ting-yuan** [comment]
```
Element: class com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl #JAVA  because: project is disposed: com.intellij.mock.MockProject@341eec3b
invalidated at: see attachment; Type element reference of class com.intellij.psi.impl.source.PsiMethodImpl #PsiMethodImpl, invalid; augmenters=[]
com.intellij.psi.PsiInvalidElementAccessException: Element: class com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl #JAVA  because: project is disposed: com.intellij.mock.MockProject@341eec3b
invalidated at: see attachment; Type element reference of class com.intellij.psi.impl.source.PsiMethodImpl #PsiMethodImpl, invalid; augmenters=[]
	at com.intellij.psi.impl.source.PsiClassReferenceType.resolveGenerics(PsiClassReferenceType.java:177)
	at com.intellij.psi.PsiClassType.equals(PsiClassType.java:110)
	at java.util.HashMap.getNode(HashMap.java:572)
	at java.util.LinkedHashMap.get(LinkedHashMap.java:440)
	at org.jetbrains.kotlin.ksp.symbol.impl.java.KSTypeReferenceJavaImpl$Companion.getCached(KSTypeReferenceJavaImpl.kt:79)
	at org.jetbrains.kotlin.ksp.symbol.impl.java.KSFunctionDeclarationJavaImpl$returnType$2.invoke(KSFunctionDeclarationJavaImpl.kt:83)
	at org.jetbrains.kotlin.ksp.symbol.impl.java.KSFunctionDeclarationJavaImpl$returnType$2.invoke(KSFunctionDeclarationJavaImpl.kt:23)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.jetbrains.kotlin.ksp.symbol.impl.java.KSFunctionDeclarationJavaImpl.getReturnType(KSFunctionDeclarationJavaImpl.kt)
	at org.jetbrains.kotlin.ksp.processor.AllFunctionsProcessor$AllFunctionsVisitor.toSignature(AllFunctionsProcessor.kt:39)
	at org.jetbrains.kotlin.ksp.processor.AllFunctionsProcessor$AllFunctionsVisitor.visitFunctionDeclaration(AllFunctionsProcessor.kt:48)
	at org.jetbrains.kotlin.ksp.processor.AllFunctionsProcessor$AllFunctionsVisitor.visitFunctionDeclaration(AllFunctionsProcessor.kt:28)
	at org.jetbrains.kotlin.ksp.symbol.impl.java.KSFunctionDeclarationJavaImpl.accept(KSFunctionDeclarationJavaImpl.kt:98)
	at org.jetbrains.kotlin.ksp.processor.AllFunctionsProcessor$AllFunctionsVisitor.visitClassDeclaration(AllFunctionsProcessor.kt:44)
	at org.jetbrains.kotlin.ksp.processor.AllFunctionsProcessor$AllFunctionsVisitor.visitClassDeclaration(AllFunctionsProcessor.kt:28)
	at org.jetbrains.kotlin.ksp.symbol.impl.kotlin.KSClassDeclarationImpl.accept(KSClassDeclarationImpl.kt:115)
	at org.jetbrains.kotlin.ksp.processor.AllFunctionsProcessor$AllFunctionsVisitor.visitFile(AllFunctionsProcessor.kt:52)
	at org.jetbrains.kotlin.ksp.processor.AllFunctionsProcessor$AllFunctionsVisitor.visitFile(AllFunctionsProcessor.kt:28)
	at org.jetbrains.kotlin.ksp.symbol.impl.kotlin.KSFileImpl.accept(KSFileImpl.kt:38)
	at org.jetbrains.kotlin.ksp.processor.AllFunctionsProcessor.process(AllFunctionsProcessor.kt:25)
	at org.jetbrains.kotlin.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:63)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:114)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:91)
	at org.jetbrains.kotlin.resolve.lazy.JvmResolveUtil.analyze(JvmResolveUtil.kt:110)
	at org.jetbrains.kotlin.resolve.lazy.JvmResolveUtil.analyzeAndCheckForErrors(JvmResolveUtil.kt:77)
	at org.jetbrains.kotlin.codegen.GenerationUtils.compileFilesUsingStandardMode(GenerationUtils.kt:169)
	at org.jetbrains.kotlin.codegen.GenerationUtils.compileFiles(GenerationUtils.kt:91)
	at org.jetbrains.kotlin.codegen.GenerationUtils.compileFiles(GenerationUtils.kt:76)
	at org.jetbrains.kotlin.codegen.GenerationUtils.compileFiles$default(GenerationUtils.kt:74)
	at org.jetbrains.kotlin.ksp.test.AbstractKotlinKSPTest.doMultiFileTest(AbstractKotlinKSPTest.kt:42)
	at org.jetbrains.kotlin.codegen.CodegenTestCase.doTest(CodegenTestCase.java:821)
	at org.jetbrains.kotlin.test.KotlinTestUtils.lambda$testWithCustomIgnoreDirective$6(KotlinTestUtils.java:791)
	at org.jetbrains.kotlin.test.MuteWithFileKt$testWithMuteInFile$1.invoke(muteWithFile.kt:41)
	at org.jetbrains.kotlin.test.KotlinTestUtils.runTestImpl(KotlinTestUtils.java:774)
	at org.jetbrains.kotlin.test.KotlinTestUtils.runTest(KotlinTestUtils.java:712)
	at org.jetbrains.kotlin.ksp.test.KotlinKSPTestGenerated.runTest(KotlinKSPTestGenerated.java:24)
	at org.jetbrains.kotlin.ksp.test.KotlinKSPTestGenerated.testAllFunctions(KotlinKSPTestGenerated.java:33)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at org.jetbrains.kotlin.test.testFramework.KtUsefulTestCase.lambda$runTest$6(KtUsefulTestCase.java:256)
	at org.jetbrains.kotlin.test.testFramework.KtUsefulTestCase.lambda$invokeTestRunnable$7(KtUsefulTestCase.java:297)
	at com.intellij.testFramework.EdtTestUtilKt.runInEdtAndWait(EdtTestUtil.kt:63)
	at org.jetbrains.kotlin.test.testFramework.KtUsefulTestCase.invokeTestRunnable(KtUsefulTestCase.java:296)
	at org.jetbrains.kotlin.test.testFramework.KtUsefulTestCase.runTest(KtUsefulTestCase.java:276)
	at org.jetbrains.kotlin.test.testFramework.KtUsefulTestCase.defaultRunBare(KtUsefulTestCase.java:311)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:18)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:13)
	at com.intellij.testFramework.EdtTestUtilKt$runInEdtAndWait$3.run(EdtTestUtil.kt:67)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:398)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

```

**Comment by ting-yuan** [comment]
Fixed in #63.
